src/ol/layer/BaseVector.js

```diff
@@ -234,13 +228,11 @@ class BaseVectorLayer extends Layer {
    * @param {import("../layer/Layer.js").State} layerState Layer state.
    */
   renderDeclutter(frameState, layerState) {
-    if (!frameState.declutterTree) {
-      frameState.declutterTree = new RBush(9);
+    const declutterGroup = this.getDeclutter();
+    if (declutterGroup in frameState.declutter === false) {
```

Comment:
Because the `declutter` option now specifies a group of layers that will be decluttered together, we use on RBush per declutter group, instead of a single global RBush for all items of all layers.